### ISSUE TITLE
gradient accumulation

### ISSUE THREAD

**Issue by bnkausik** [karpathy/nanoGPT#605]
in train.py line 301:

`loss = loss / gradient_accumulation_steps # scale the loss to account for gradient accumulation**`

should it be 

`loss = loss / (micro_step+1)  # scale the loss to account for gradient accumulation**
`

since we've accumulated (micro_step+1) gradients at a given iteration

thanks for clarfying
